.github/workflows/generative-tests.yml

```
@@ -46,10 +46,18 @@ jobs:
             echo
 
             # Do the actual testing
-            just test tests/generative/test_query_model.py::test_query_model
+            just test tests/generative/test_test_query_model.py
+            rc=%?
+            if [ $rc -ne 0] && [ $rc -ne 6 ]; then
+                break
```

This isn't quite the right behaviour: we don't want to break from the loop in case of error because we'll end up exiting cleanly and then the test will appear to have passed; instead we want to exit with whatever status the test command exited with:
```suggestion
                exit $rc
```